### PR TITLE
Fixes for libntech without GPL code

### PR DESCRIPTION
**Pull request by vpodzime**
Merge together:
https://github.com/cfengine/core/pull/5282
https://github.com/cfengine/libntech/pull/186

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9591/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9591/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9591/

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9602/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9602/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9602/

**Comment by vpodzime** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9607/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9607/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9607/

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9608/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9608/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9608/

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics again, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9611/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9611/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9611/

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @olehermanse can review this?

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9632/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9632/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9632/

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9640/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9640/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9640/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9645/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9645/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9645/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9651/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9651/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9651/

File: tests/unit/getopt_test.c [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    while ((c = getopt_long(argc, argv, "no::r:", OPTIONS, NULL)) != -1)
+    {
+        switch (c)
+        {
+        case 'n': /* no argument */
+            SeqAppend(seq, xstrdup("none"));
+            break;
+
+        case 'o': /* optional argument */
+            if (OPTIONAL_ARGUMENT_IS_PRESENT)
```
Comment (larsewi) [inline tests/unit/getopt_test.c:38]: 🚀

**Review by larsewi — APPROVED** [review]

**Review by olehermanse — APPROVED** [review]

File: configure.ac [hunk 2]
```diff
@@ -1222,10 +1190,8 @@ AC_CHECK_DECLS([fchmodat], [], [], [[#define _GNU_SOURCE 1
                                     #include <sys/stat.h>]])
 AC_CHECK_DECLS([readlinkat], [], [], [[#define _GNU_SOURCE 1
                                        #include <unistd.h>]])
-AC_REPLACE_FUNCS(openat fstatat fchownat fchmodat readlinkat)
 
 AC_CHECK_DECLS([log2], [], [], [[#include <math.h>]])
-AC_REPLACE_FUNCS(log2)
```
Comment (craigcomstock) [inline configure.ac:1228]: @vpodzime can we do some cleanup of these checks in core, nova, enterprise as well? How did you go about knowing which ones to remove?

Comment (craigcomstock) [reply configure.ac:1228]: Oops. I thought this was a libntech PR ;) this is core. Awesome that you removed all those checks. 🥇

**Pull request merged**